client/packages/system/src/Patient/JsonForms/components/IdGenerator.tsx

```diff
@@ -276,7 +284,11 @@ const UIComponent = (props: ControlProps) => {
         />
 
         <Box>
-          <Button disabled={error} onClick={generate} variant="outlined">
+          <Button
+            disabled={error || !canGenerate}
+            onClick={generate}
+            variant="outlined"
+          >
```

Comment:
Sorry think you missed these comments:
https://github.com/openmsupply/open-msupply/issues/655#issuecomment-1334420579

Our concern were that there might be a legit reason why you want to regenerate after the first safe, e.g. directly after you created a user.

Reply:
Yeah, I saw that. Planning to address it shortly...

Reply:
Okay, have updated: 3d37523d0

Added comment block explaining options for regenerating:
```ts
  /*
  Regeneration behaviour:
  - By default, after ID is first saved, a confirmation will be displayed
    whenever the user subsequently clicks the "Generate" button
  - This can be suppressed by setting the "confirmRegenerate" option to `false`
    (default `true`)
  - To prevent the ID from *ever* being regenerated after first save, set the
    "preventRegenAfterSave" option to `true` (default `false`)
  */
```